api/agents.go

```diff
@@ -2,18 +2,29 @@ package api
 
 // AgentRegisterRequest is a call to register on the Buildkite Agent API
 type AgentRegisterRequest struct {
-	Name               string   `json:"name"`
-	Hostname           string   `json:"hostname"`
-	OS                 string   `json:"os"`
-	Arch               string   `json:"arch"`
-	ScriptEvalEnabled  bool     `json:"script_eval_enabled"`
-	IgnoreInDispatches bool     `json:"ignore_in_dispatches"`
-	Priority           string   `json:"priority,omitempty"`
-	Version            string   `json:"version"`
-	Build              string   `json:"build"`
-	Tags               []string `json:"meta_data"`
-	PID                int      `json:"pid,omitempty"`
-	MachineID          string   `json:"machine_id,omitempty"`
+	Name               string        `json:"name"`
+	Hostname           string        `json:"hostname"`
+	OS                 string        `json:"os"`
+	Arch               string        `json:"arch"`
+	ScriptEvalEnabled  bool          `json:"script_eval_enabled"`
+	IgnoreInDispatches bool          `json:"ignore_in_dispatches"`
+	Priority           string        `json:"priority,omitempty"`
+	Version            string        `json:"version"`
+	Build              string        `json:"build"`
+	Tags               []string      `json:"meta_data"`
+	PID                int           `json:"pid,omitempty"`
+	MachineID          string        `json:"machine_id,omitempty"`
+	FeatureUsage       *FeatureUsage `json:"feature_usage"`
+}
+
+type FeatureUsage struct {
+	TracingBackend             string   `json:"tracing_backend"`
```

Comment:
Should this be omitted when empty?

```suggestion
	TracingBackend             string   `json:"tracing_backend,omitempty"`
```

Reply:
ah yes! it should, it's been so long since i serialised a struct, i completely forgot.

Reply:
actually, thinking on this, i think it makes sense to leave that as `""` if there's no tracing backend configured - i think it's marginally clearer in that case that tracing is disabled?

the other option would be to have two things in here, `tracing_enabled bool` and `tracing_backend bool omitempty`

Reply:
Or, `open_tracing_enabled` and `open_telemetry_enabled` bools.
I think that's my favourite.